packages/splits-sdk/src/client/splitV2.ts

```diff
@@ -627,24 +628,31 @@ export class SplitV2Client extends SplitV2Transactions {
   async getSplitMetadata(splitAddress: Address): Promise<{ split: SplitV2 }> {
     this._requirePublicClient()
 
-    const [updateLogs, createLogs, owner] = await Promise.all([
-      this._publicClient?.getLogs({
-        address: splitAddress,
-        event: splitUpdatedEvent,
-        strict: true,
-      }),
+    const [createLogs, owner] = await Promise.all([
       this._publicClient?.getLogs({
         event: splitCreatedEvent,
+        address: [
+          getSplitV2FactoryAddress(this._chainId, SplitV2Type.Pull),
+          getSplitV2FactoryAddress(this._chainId, SplitV2Type.Push),
+        ],
         args: {
           split: splitAddress,
         },
         strict: true,
+        fromBlock: getSplitV2FactoriesStartBlock(this._chainId),
       }),
       this.owner(splitAddress),
     ])
 
     if (!createLogs) throw new Error('Split not found')
 
+    const updateLogs = await this._publicClient?.getLogs({
+      address: splitAddress,
+      event: splitUpdatedEvent,
+      strict: true,
+      fromBlock: createLogs[0].blockNumber,
```

Comment:
ah nice, i was gonna ask why you moved this request to be afterwards. makes sense